tests/common/igt.cpp

```diff
@@ -0,0 +1,329 @@
+/*
+* Copyright:
+* code of this file copied/modified from
+* https://anongit.freedesktop.org/git/xorg/app/intel-gpu-tools.git
+* commit: 0a0eb5d48bab8df3e2fd530fb631e1478c9c9215
+*/
+
+#include <inttypes.h>
+#include <sys/stat.h>
+#include <sys/mount.h>
+#include <dirent.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <igt.h>
+
+const char *kmstest_pipe_name(int pipe) {
+  const char *str[] = {"A", "B", "C"};
+
+  if (pipe == -1)
+    return "None";
+
+  if (pipe > 2)
+    return "invalid";
+
+  return str[pipe];
+}
+
+const char *igt_debugfs_mount(void) {
+  struct stat st;
+
+  if (stat("/debug/dri", &st) == 0)
+    return "/debug";
+
+  if (stat("/sys/kernel/debug/dri", &st) == 0)
+    return "/sys/kernel/debug";
+
+  return "/sys/kernel/debug";
+}
+
+static bool __igt_debugfs_init(igt_debugfs_t *debugfs) {
+  struct stat st;
+  int n;
+
+  strcpy(debugfs->root, igt_debugfs_mount());
+  for (n = 0; n < 16; n++) {
+    int len = sprintf(debugfs->dri_path, "%s/dri/%d", debugfs->root, n);
+    sprintf(debugfs->dri_path + len, "/i915_error_state");
+    if (stat(debugfs->dri_path, &st) == 0) {
+      debugfs->dri_path[len] = '\0';
+      return true;
+    }
+  }
+
+  debugfs->dri_path[0] = '\0';
+  return false;
+}
+
+static igt_debugfs_t *__igt_debugfs_singleton(void) {
+  static igt_debugfs_t singleton;
+  static bool init_done = false;
+
+  if (init_done)
+    return &singleton;
+
+  if (__igt_debugfs_init(&singleton)) {
+    init_done = true;
+    return &singleton;
+  } else {
+    return NULL;
+  }
+}
+
+FILE *igt_debugfs_fopen(const char *filename, const char *mode) {
+  char buf[1024];
+
+  igt_debugfs_t *debugfs = __igt_debugfs_singleton();
+
+  if (!debugfs)
+    return NULL;
+
+  sprintf(buf, "%s/%s", debugfs->dri_path, filename);
+  return fopen(buf, mode);
+}
+
+int igt_debugfs_open(const char *filename, int mode) {
+  char buf[1024];
+  igt_debugfs_t *debugfs = __igt_debugfs_singleton();
+
+  if (!debugfs)
+    return -1;
+
+  sprintf(buf, "%s/%s", debugfs->dri_path, filename);
+  return open(buf, mode);
+}
+
+bool igt_require_pipe_crc(void) {
+  const char *cmd = "pipe A none";
+  FILE *ctl;
+  size_t written;
+  int ret;
+
+  ctl = igt_debugfs_fopen("crtc-0/crc/control", "r+");
+  if (!ctl) {
+    ctl = igt_debugfs_fopen("i915_display_crc_ctl", "r+");
+    if (ctl)
+      printf("No display_crc_ctl found, kernel too old\n");
+    return false;
```

Comment:
You are missing brackets here (if (ctl) condition), u will always return false here otherwise.